diff --git a/include/base/cef_template_util.h b/include/base/cef_template_util.h
index 6661fc035..5d6c9cdf1 100644
--- a/cef/include/base/cef_template_util.h
+++ b/cef/include/base/cef_template_util.h
@@ -166,15 +166,7 @@ struct priority_tag<0> {};
     defined(CR_USE_FALLBACKS_FOR_GCC_WITH_LIBCXX)
 template <typename T>
 struct is_trivially_copyable {
-// TODO(danakj): Remove this when android builders are all using a newer version
-// of gcc, or the android ndk is updated to a newer libc++ that does this for
-// us.
-#if _GNUC_VER >= 501
   static constexpr bool value = __is_trivially_copyable(T);
-#else
-  static constexpr bool value =
-      __has_trivial_copy(T) && __has_trivial_destructor(T);
-#endif
 };
 #else
 template <class T>
