diff --git a/include/views/cef_browser_view.h b/include/views/cef_browser_view.h
index e8b4218ff..da06146b3 100644
--- a/cef/include/views/cef_browser_view.h
+++ b/cef/include/views/cef_browser_view.h
@@ -39,9 +39,10 @@
 #pragma once
 
 #include "include/cef_browser.h"
-#include "include/views/cef_browser_view_delegate.h"
 #include "include/views/cef_view.h"
 
+class CefBrowserViewDelegate;
+
 ///
 /// A View hosting a CefBrowser instance. Methods must be called on the browser
 /// process UI thread unless otherwise indicated.
diff --git a/include/views/cef_browser_view_delegate.h b/include/views/cef_browser_view_delegate.h
index 262c14521..5da26949c 100644
--- a/cef/include/views/cef_browser_view_delegate.h
+++ b/cef/include/views/cef_browser_view_delegate.h
@@ -40,9 +40,7 @@
 
 #include "include/cef_client.h"
 #include "include/views/cef_view_delegate.h"
-
-class CefBrowser;
-class CefBrowserView;
+#include "include/views/cef_browser_view.h"
 
 ///
 /// Implement this interface to handle BrowserView events. The methods of this
diff --git a/include/views/cef_view.h b/include/views/cef_view.h
index 9bfd1f47b..55bcb17ce 100644
--- a/cef/include/views/cef_view.h
+++ b/cef/include/views/cef_view.h
@@ -38,13 +38,14 @@
 #define CEF_INCLUDE_VIEWS_CEF_VIEW_H_
 #pragma once
 
-#include "include/views/cef_view_delegate.h"
+#include "include/cef_base.h"
 
 class CefBrowserView;
 class CefButton;
 class CefPanel;
 class CefScrollView;
 class CefTextfield;
+class CefViewDelegate;
 class CefWindow;
 
 ///
diff --git a/include/views/cef_view_delegate.h b/include/views/cef_view_delegate.h
index 23878323e..96c30b8b9 100644
--- a/cef/include/views/cef_view_delegate.h
+++ b/cef/include/views/cef_view_delegate.h
@@ -39,8 +39,7 @@
 #pragma once
 
 #include "include/cef_base.h"
-
-class CefView;
+#include "include/views/cef_view.h"
 
 ///
 /// Implement this interface to handle view events. All size and position values
diff --git a/tests/cefsimple/simple_app.cc b/tests/cefsimple/simple_app.cc
index 72556cf5a..01ed859b1 100644
--- a/cef/tests/cefsimple/simple_app.cc
+++ b/cef/tests/cefsimple/simple_app.cc
@@ -8,7 +8,7 @@
 
 #include "include/cef_browser.h"
 #include "include/cef_command_line.h"
-#include "include/views/cef_browser_view.h"
+#include "include/views/cef_browser_view_delegate.h"
 #include "include/views/cef_window.h"
 #include "include/wrapper/cef_helpers.h"
 #include "tests/cefsimple/simple_handler.h"
diff --git a/tests/ceftests/test_handler.cc b/tests/ceftests/test_handler.cc
index 3ebf4ae16..d7caa27e8 100644
--- a/cef/tests/ceftests/test_handler.cc
+++ b/cef/tests/ceftests/test_handler.cc
@@ -8,7 +8,7 @@
 #include "include/base/cef_logging.h"
 #include "include/cef_command_line.h"
 #include "include/cef_stream.h"
-#include "include/views/cef_browser_view.h"
+#include "include/views/cef_browser_view_delegate.h"
 #include "include/views/cef_window.h"
 #include "include/wrapper/cef_closure_task.h"
 #include "include/wrapper/cef_stream_resource_handler.h"
